checker/src/main/java/org/checkerframework/checker/calledmethods/builder/AutoValueSupport.java

```diff
@@ -0,0 +1,448 @@
+                            requiredProperties, avBuilderSetterNames);
+            // only add the new @CalledMethods annotation if there is not already a @CalledMethods
+            // annotation present
+            AnnotationMirror possibleBuildAnnotations =
+                    t.getReceiverType()
+                            .getAnnotationInHierarchy(
+                                    atypeFactory
+                                            .getQualifierHierarchy()
+                                            .getTopAnnotation(newCalledMethodsAnno));
+            if (possibleBuildAnnotations == null) {
+                t.getReceiverType().addAnnotation(newCalledMethodsAnno);
+            }
+        }
+    }
+
+    @Override
+    public void handlePossibleToBuilder(AnnotatedExecutableType t) {
+
+        AnnotatedTypeMirror returnType = t.getReturnType();
+        ExecutableElement element = t.getElement();
+
+        String methodName = element.getSimpleName().toString();
+
+        // make sure the method is toBuilder
+        if ("toBuilder".equals(methodName)) {
+
+            TypeElement enclosingElement = (TypeElement) element.getEnclosingElement();
+            TypeMirror superclass = enclosingElement.getSuperclass();
+
+            if (ElementUtils.hasAnnotation(
+                            enclosingElement, getAutoValuePackageName() + ".AutoValue")
+                    && element.getModifiers().contains(Modifier.ABSTRACT)) {
+                handleToBuilderType(returnType, returnType.getUnderlyingType(), enclosingElement);
+            }
+
+            if (superclass.getKind() != TypeKind.NONE) {
+                TypeElement superElement = TypesUtils.getTypeElement(superclass);
+                if (ElementUtils.hasAnnotation(
+                        superElement, getAutoValuePackageName() + ".AutoValue")) {
+                    handleToBuilderType(returnType, returnType.getUnderlyingType(), superElement);
+                }
+            }
+        }
+    }
+
+    /**
+     * Add, to {@code type}, a CalledMethods annotation with all required methods called. The type
+     * can be the return type of toBuilder or the corresponding generated "copy" constructor.
+     *
+     * @param type type to update
+     * @param builderType type of abstract @AutoValue.Builder class
+     * @param classElement corresponding AutoValue class
+     */
+    private void handleToBuilderType(
+            AnnotatedTypeMirror type, TypeMirror builderType, Element classElement) {
+        Element builderElement = TypesUtils.getTypeElement(builderType);
+        Set<String> avBuilderSetterNames = getAutoValueBuilderSetterMethodNames(builderElement);
+        List<String> requiredProperties =
+                getAutoValueRequiredProperties(classElement, avBuilderSetterNames);
+        AnnotationMirror calledMethodsAnno =
+                createCalledMethodsForAutoValueProperties(requiredProperties, avBuilderSetterNames);
```

Comment:
I followed the former suggestion, but I disagree with the latter. Because there is no need for such a method outside the implementation of a class that is implementing `BuilderFrameworkSupport`, adding it to the interface would clutter the interface and make implementation of new builder framework support classes more annoying. (I agree that it's common to all builder supports, but I don't think it's helpful to proscribe e.g. the inputs for such a method, since different builder frameworks might use different information to decide on the required properties.)